tests/cycling-robot.cpp

```diff
@@ -0,0 +1,218 @@
+      stage_space, actuation_matrix, constraint_models, proximal_settings);
+    return dynamics::IntegratorSemiImplEulerTpl<double>(ode, dt);
+  }
+
+  auto makeStage(double t, double dt, pinocchio::Model const& model)
+  {
+    auto const rcost = makeCost(t, model);
+    auto const dynModel = makeDynamicsModel(t, dt, model);
+
+    return StageModelTpl<double>(rcost, dynModel);
+  }
+
+} // namespace
+
+BOOST_AUTO_TEST_CASE(test_cycling_robot) {
+
+  auto const model = SimpleModel();
+  auto data = pinocchio::Data(model);
+
+  auto const nq = model.nq;
+  auto const nv = model.nv;
+  auto const nx = nq + nv;
+  auto const nu = ::nu(model);
+  auto constexpr nsteps = 10;
+  auto constexpr dt = 0.02;
+
+  std::cout << "nq: " << nq << ", nv: " << nv << ", nx: " << nx << ", nu: " << nu << std::endl;
+
+  auto q = pinocchio::neutral(model);
+  auto vq = Eigen::VectorXd::Zero(nv).eval();
+  auto x = Eigen::VectorXd::Zero(nx).eval();
+  x << q, vq;
+
+  pinocchio::computeAllTerms(model, data, q, vq);
+  pinocchio::updateFramePlacements(model, data);
+
+  auto const term_cost = makeCost(0., model);
+  auto problem = TrajOptProblemTpl<double>(x, nu, stage_space_t(model), term_cost);
+
+  for (auto i = 0; i < nsteps; ++i)
+    problem.addStage(makeStage(0., dt, model));
+
+  auto constexpr TOL = 1e-7;
+  auto constexpr max_iters = 100u;
+  auto constexpr mu_init = 1e-8;
+  auto constexpr verbosity = QUIET;
+
+  auto ddp = SolverProxDDPTpl<double>(TOL, mu_init, max_iters, verbosity);
+  ddp.rollout_type_ = RolloutType::LINEAR;
+  ddp.sa_strategy_ = StepAcceptanceStrategy::LINESEARCH_NONMONOTONE;
+  ddp.filter_.beta_ = 1e-5;
+  ddp.force_initial_condition_ = true;
+  ddp.reg_min = 1e-6;
+  ddp.linear_solver_choice = LQSolverChoice::SERIAL;
+
+  ddp.setup(problem);
+
+  auto converged = ddp.run(problem);
+  BOOST_CHECK(converged);
+
+  for (auto t = 0.; t < 5.; t += dt)
+  {
+    auto const t0 = std::chrono::steady_clock::now();
+    auto const x = ddp.results_.xs[1];
+    q = x.head(nq);
+    vq = x.tail(nv);
+    pinocchio::computeAllTerms(model, data, q, vq);
+    pinocchio::updateFramePlacements(model, data);
+
+    auto const stage = makeStage(t, dt, model);
+    problem.replaceStageCircular(stage);
+    ddp.cycleProblem(problem, stage.createData());
+    problem.term_cost_ = makeCost(t + dt, model);
+    ddp.workspace_.problem_data.term_cost_data = problem.term_cost_->createData();
+    problem.setInitState(x);
+
+    auto converged = ddp.run(problem);
+    BOOST_CHECK(converged);
+    auto const [expected, support] = height(std::max(0., t - (nsteps-1) * dt));
+    auto const& actual = data.oMf[model.getFrameId("link")];
+    BOOST_CHECK((actual.inverse() * expected).isIdentity(2e-3));
+    BOOST_CHECK_SMALL(actual.translation()[2] - expected.translation()[2], 2e-3);
+
+    auto const tf = std::chrono::steady_clock::now();
+    std::cout << "Time: " << std::chrono::duration<double, std::milli>(tf - t0).count() << " ms" << std::endl;
```

Comment:
You should use `fmt::print`/[`fmt::printf`](https://fmt.dev/11.1/api/#safe-printf) instead of the stuff from the dreadful `<iostream>`. `aligator` already links publicly to `fmt` so it's no issue.